### PR TITLE
Context instead of Activity in SiteSettingsInterface

### PR DESCRIPTION
**Pull request by hypest**
This way, SiteSettingsInterface can be used by Services as well.

No new functionality added.

To test:
* Try to change the site settings via the app. It should work and the remote site be updated as well.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsInterface.java [hunk 1]
```diff
@@ -226,9 +227,9 @@ protected void finalize() throws Throwable {
 
     public void saveSettings() {
         SiteSettingsTable.saveSettings(mSettings);
-        siteSettingsPreferences(mActivity).edit().putString(LANGUAGE_PREF_KEY, mSettings.language).apply();
-        siteSettingsPreferences(mActivity).edit().putInt(DEF_CATEGORY_PREF_KEY, mSettings.defaultCategory).apply();
-        siteSettingsPreferences(mActivity).edit().putString(DEF_FORMAT_PREF_KEY, mSettings.defaultPostFormat).apply();
+        siteSettingsPreferences(mContext).edit().putString(LANGUAGE_PREF_KEY, mSettings.language).apply();
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsInterface.java]: I know this is pre-existing code, but how about using the builder pattern here?

```
siteSettingsPreferences(mContext).edit()
        .putString(LANGUAGE_PREF_KEY, mSettings.language)
        .putInt(DEF_CATEGORY_PREF_KEY, mSettings.defaultCategory)
        .putString(DEF_FORMAT_PREF_KEY, mSettings.defaultPostFormat)
        .apply();
```

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsInterface.java [hunk 2]
```diff
@@ -939,11 +940,13 @@ public void run() {
     }
 
     protected void notifyFetchErrorOnUiThread(final Exception error) {
-        if (mActivity == null || mActivity.isFinishing() || mListener == null) {
+        if (mContext == null
+                || (mContext instanceof Activity && ((Activity) mContext).isFinishing())
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsInterface.java]: Do we really need to check `isFinishing()` here? That seems like something the caller should deal with.

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsInterface.java (hunk 1 above)
Comment (hypest) [reply WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsInterface.java]: Makes sense, done in c97620d.

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsInterface.java (hunk 2 above)
Comment (hypest) [reply WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsInterface.java]: I was only adapting the existing code without changing the logic and yeah, that check is awkward. I removed it with 0247fd6, letting the listener decide whether the Activity (if any) needs to be checked for finishing. It seems that almost all listeners are already checking for `isAdded` so, no new checks are needed.

**Comment by hypest** [comment]
I addressed your comments @nbradbury , ready for another pass, thanks!

**Comment by nbradbury** [comment]
Looks good! :shipit:

**Pull request merged**